homeassistant/util/dt.py

```diff
@@ -267,36 +267,61 @@ def parse_time(time_str: str) -> dt.time | None:
         return None
 
 
-def get_age(date: dt.datetime) -> str:
+def get_age(date: dt.datetime, is_future: bool = False, depth: int = 1) -> str:
```

Comment:
Given the name of the function, it seems very unnatural to add a parameter `is_future` to it; I would expect it to return a negative age for a time in the future.

Maybe split this up in multiple functions:
- get_age, which returns the time elapsed since a time in the past
- a new function, maybe named something like `get_remaining_time` which returns the remaining time to a time in the future
- A helper functions which formats the time delta calculated by `get_age` and the new function as a string

Reply:
What about:

* Rename this to `get_diff()`
* Change `is_future` to an enum, with values like `DiffType.PAST_ONLY`, `DiffType.FUTURE_ONLY`, `DiffType.ANY` -- this makes the code much easier to read than trying to figure out what a bool means in this context
* Re-add `get_age` for backwards-compat that just calls `get_diff(date, DiffType.PAST_ONLY, 1)`

Note: There may not be value in having `ANY`, but being an enum at least opens this possibility.  (I feel like I've needed a way to do a past-or-future date before, but I'm struggling to think of a good use case now)

Reply:
Your suggestion is fine 👍 

> There may not be value in having ANY, but being an enum at least opens this possibility.

Let's not add functionality which we are not sure is needed.

Reply:
It's been a while, but I think the is_future was meant to be more like an "allow future" flag.

Reply:
Yes, but then please come up with a good name for a bool which clearly communicates how the behavior changes. I don't think `allow_future` is good because it's not clear setting it to `True` means a time in the past is no longer allowed.

Hence my proposal to leave the behavior of `get_age` unchanged and add a new function `get_remaining_time` which returns remaining time until some time in the future.